app/models/course/group_user.rb

```diff
@@ -0,0 +1,18 @@
+class Course::GroupUser < ActiveRecord::Base
+  stampable
+  belongs_to :user, inverse_of: :group_users
+  belongs_to :group, class_name: Course::Group.name, inverse_of: :group_users,
+                     foreign_key: :course_group_id
+
+  validate :user_and_group_in_same_course
+
+  enum role: { normal: 0, manager: 1 }
+
+  private
+
+  def user_and_group_in_same_course #:nodoc:
+    unless user.courses.include?(group.course)
```

Comment:
Use a guard clause instead of wrapping the code inside a conditional expression.